docs/transactions/transaction-attributes.md

```diff
@@ -0,0 +1,36 @@
+---
+sidebar_position: 30
+title: Transaction Attributes
+---
+
+This guide provides an in-depth understanding of the transaction attributes in Macrometa's Global Data Network (GDN). Each attribute plays a critical role in ensuring efficient, secure, and reliable transactions.
+
+## `allowImplicit` Attribute
+
+The `allowImplicit` attribute determines if a transaction can read from collections not declared at the start. By default, `allowImplicit` is set to `true`, allowing this flexibility. However, setting it to `false` restricts the transaction to only the declared collections, enhancing isolation but requiring upfront knowledge of all needed collections.
+
+## `collections` Attribute
+
+The `collections` attribute is central to transaction management. It specifies which collections the transaction interacts with and in what manner:
+
+- `read`: Lists collections accessed only for reading, which is recommended to prevent deadlocks. This is optional if `allowImplicit` is set to `true`.
+- `write`: Includes collections where data might be read or modified.
+- `exclusive`: Specifies collections for exclusive write access, preventing concurrent write operations. It essentially "locks" the collection.
+
+This attribute ensures data integrity and governs access levels to collections within a transaction. However, it can lead to deadlocks if a collection that is read from within the transaction is not declared in the read sub-attribute.
+
+## `lockTimeout` Attribute
+
+`lockTimeout` sets a time limit in seconds for how long a transaction waits to acquire a lock on a collection. It's particularly relevant for transactions requiring exclusive access. A well-set `lockTimeout` can prevent transactions from stalling indefinitely, especially when set to 0, causes the transaction to wait indefinitely. Default lock timeout is 900 seconds.
```

Comment:
"A well-set `lockTimeout` can prevent transactions from stalling indefinitely, especially when set to 0, causes the transaction to wait indefinitely."

Isn't this sentence contradictory? First it suggests that setting a reasonable `lockTimeout` value can avoid transactions from getting stuck indefinitely. Then it says that setting `lockTimeout` to 0 causes the transaction to wait indefinitely, which is contrary to the idea of preventing transactions from stalling.